tools/build_docker.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+cur_path=$(dirname $(readlink -f $0))
+cd $cur_path/../
+
+#convert md to ipynb
+./tools/convert-markdown-into-ipynb-and-test.sh
+
+#get submodule
+git submodule update --init --recursive
+cd paddle && paddle_version=`git describe --abbrev=0 --tags` && cd ..
+if [ $? -ne 0 ]; then
+	echo 1>&2 "get paddle version error"
+	exit 1
+fi
+
+#generate docker file
+if [ ${USE_MIRROR} ]; then
+  MIRROR_UPDATE="sed 's@http:\/\/archive.ubuntu.com\/ubuntu\/@mirror:\/\/mirrors.ubuntu.com\/mirrors.txt@' -i /etc/apt/sources.list && \\"
+else
+  MIRROR_UPDATE="\\"
+fi
+
+mkdir -p build
+cat > build/Dockerfile <<EOF
+FROM paddlepaddle/paddle:${paddle_version}
+MAINTAINER PaddlePaddle Authors <paddle-dev@baidu.com>
+
+RUN apt-get install locales 
+RUN localedef -f UTF-8 -i en_US en_US.UTF-8
+
+RUN ${MIRROR_UPDATE}
+	apt-get -y install gcc && \
+	apt-get -y clean
+
+RUN pip install -U matplotlib jupyter numpy requests
+
+COPY . /book
+RUN rm -rf /book/build
+
+EXPOSE 8888
+CMD ["jupyter", "notebook", "--ip=0.0.0.0", "/book/"]
```

Comment:
according to https://github.com/ipython/ipython/issues/7062 change to `CMD ["sh", "-c", "jupyter", "notebook", "--ip=0.0.0.0", "/book/"]` solves the kernel dead problem